packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts

```diff
@@ -324,6 +324,9 @@ export class Schedule extends Resource implements ISchedule {
     const flexibleTimeWindow = props.timeWindow ?? TimeWindow.off();
 
     this.validateTimeFrame(props.start, props.end);
+    if (props.scheduleName && !Token.isUnresolved(props.scheduleName) && props.scheduleName.length > 64) {
+      throw new Error(`schedule name '${props.scheduleName}' exceeds 64 characters in length`);
```

Comment:
It would be user-friendly and good to display the length.

```suggestion
      throw new Error(`scheduleName cannot be longer than 64 characters, got: ${props.scheduleName.length}`);
```